content/ko/docs/topics/charts_hooks.md

```diff
@@ -0,0 +1,205 @@
+   준비가 될 때까지 `post-install` 훅을 실행하지 않는다.
+10. 라이브러리는 `post-install` 훅을 실행한다. (훅 리소스 로딩)
+11. 라이브러리는 훅이 "Ready" 될 때까지 기다린다.
+12. 라이브러리는 릴리스 객체(및 다른 데이터)를 클라이언트에 반환한다.
+13. 클라이언트가 종료된다.
+
+훅이 준비될 때까지 기다린다는 것은 무엇을 의미하는가? 이것은 
+훅에 선언된 리소스에 따라 다르다. 리소스가 `Job` 또는 `Pod` 종류인 경우 
+헬름은 성공적으로 실행 완료될 때까지 기다린다. 훅이 실패하면 릴리스는 실패하게 된다. 
+이것은 _블로킹(blocking) 작업_이므로 작업이 실행되는 
+동안 헬름 클라이언트는 일시정지될 것이다.
+
+다른 모든 종류의 경우, 쿠버네티스가 리소스를 로드(추가 또는 업데이트)하였다고 
+표시하는 즉시, 리소스는 "Ready" 로 간주된다.
+훅에 많은 리소스가 선언되면 리소스들은 연속적으로 실행된다.
+훅 가중치(아래 참조)가 있으면, 가중치 순서대로 실행된다.
+헬름 3.2.0 부터 같은 가중치를 가진 훅 리소스들은 훅이 없는 일반 리소스와 
+같은 순서로 설치된다. 그밖의 경우에는 순서가 보장되지 않는다. (헬름 2.3.0 
+이상에서는 알파벳순으로 정렬된다. 그러나 이 동작은 바인딩으로 간주되지 않으며 
+향후 변경될 수 있다.) 여기까지가 훅 가중치 추가시 고려할 사항이며,
+만약 가중치가 중요하지 않다면 `0` 으로 설정하는 것이 
+좋다.
+
+### 해당 릴리스에 의해 관리되지 않는 훅 리소스
+
+훅이 생성하는 리소스는 현재 릴리스의 일부로 추적되거나 
+관리되지 않는다. 헬름은 훅이 준비 상태까지 도달했는지를 확인했다면
+훅 리소스를 그대로 남겨둔다. 향후 헬름 3에 릴리스 삭제시 
+훅 리소스에 대한 가비지 수집 기능이 추가될 수 있으므로, 
+삭제되면 안되는 모든 훅 리소스에 `helm.sh/resoure-policy: keep` 어노테이션을 
+추가해야 한다.
+
+실질적으로, 이는 훅 리소스를 만드는 경우 `helm uninstall` 으로 
+리소스를 제거할 수 없음을 의미한다. 이러한 리소스를 
+삭제하려면 훅 템플릿 파일에 [사용자 지정 `helm.sh/hook-delete-policy` 어노테이션을 추가](#훅-삭제-정책)하거나 
+작업(Job) 리소스의 
+[time to live(TTL) 필드를 설정](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/)해야 
+한다.
+
+## 훅 작성하기
+
+훅은 `metadata` 섹션에 특별한 어노테이션이 있는 
+쿠버네티스 매니페스트 파일이다. 템플릿 파일이기 때문에 
+`.Values`, `.Release` 및 `.Template` 을 
+읽기를 포함한 모든 일반 템플릿 기능을 사용할 수 있다.
+
+예를 들어 `templates/post-install-job.yaml` 에 저장된 템플릿은
+`post-install` 에서 실행할 작업을 선언한다.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ .Release.Name }}"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: post-install-job
+        image: "alpine:3.3"
+        command: ["/bin/sleep","{{ default "10" .Values.sleepyTime }}"]
+
+```
+
+다음의 어노테이션으로 이 템플릿을 훅으로 만든다.
+
+```yaml
+annotations:
+  "helm.sh/hook": post-install
+```
+
+하나의 리소스는 여러 개의 훅으로 구현할 수 있다.
+
+```yaml
+annotations:
+  "helm.sh/hook": post-install,post-upgrade
+```
+
+마찬가지로, 주어진 훅을 구현하는 다양한 리소스의 수에는 
+제한이 없다. 예를 들어 시크릿과 컨피그맵을 둘다
+pre-install 훅으로 선언할 수 있다.
+
+하위 차트에 훅이 선언되면 해당 훅도 평가된다.
+최상위 차트가 하위 차트에서 선언한 훅을 비활성화 할 수 있는 방법은 없다.
```

Comment:
The English here says "There is no way for a top-level chart to disable the hooks declared by subcharts."

The google translate for this line says "There is no way to disable hooks declared by the top chart in the subchart."

To clarify, the English says "if the hooks are declared by a subchart, the top-level chart cannot disable those hooks." Is this meaning clear in the Korean translation? It sounds like it might be saying that subcharts cannot disable hooks from the top-level chart (which is wrong).

Reply:
The meaning of translation is correct though I'm not sure that this translated korean line is good for re-translation into english.
I think it may be mis-translation of Google translator.

The structure of this line is following:
"최상위 차트"가 하위 차트에서 선언한 "훅"을 비활성화 할 수 있는" 방법은 없다.
The subject is "최상위 차트" and object is "훅".

So English expression of korean line is:
"There is no way for "최상위 차트" to disable the hooks declared by "하위 차트".

In this case, we can clarify by adding a comma after subject.

A : 최상위 차트가 하위 차트에서 선언한 훅을 비활성화 할 수 있는 방법은 없다.
B : 최상위 차트가, 하위 차트에서 선언한 훅을 비활성화 할 수 있는 방법은 없다.

You can confirm correctness by using google translator about "sentence B"

And I will change into B sentence if you think it's better than expression I translate. :D

@jmnote @sunminjeon @singlemocha Please comment on the above.

Reply:
First, the quality of google translation for Korean is not so good. bluefriday's translation is pretty good. But, I want to suggest another one.

```suggestion
최상위 차트가 하위 차트에 선언된 훅을 비활성화할 수 있는 방법은 없다.
```

Reply:
I will change for your suggestion. Thanks :D